packages/gatsby-source-drupal/src/utils.js

```diff
@@ -78,11 +80,16 @@ const handleReferences = (
     })
 
     delete node.drupal_relationships
-    referencedNodesLookup.set(node, referencedNodes)
+    referencedNodesLookup.set(node.id, referencedNodes)
     if (referencedNodes.length) {
       const nodeFieldName = `${node.internal.type}___NODE`
       referencedNodes.forEach(nodeID => {
-        const referencedNode = getNode(nodeID)
+        let referencedNode
+        if (mutateNode) {
+          referencedNode = getNode(nodeID)
```

Comment:
On the initial sourcing, it does make sense to directly mutate nodes as none of them have been created yet.